azure-quantum/azure/quantum/qiskit/qir/builder.py

```diff
@@ -0,0 +1,106 @@
+##
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+##
+
+from dis import Instruction
+from typing import Union, Iterable
+
+from pyqir_generator import QirBuilder
+from qiskit.circuit.classicalregister import ClassicalRegister
+from qiskit.circuit.quantumregister import QuantumRegister
+
+
+INSTRUCTIONS = [
+    "cx",
+    "cz",
+    "h",
+    "m",
+    "reset",
+    "rx",
+    "ry",
+    "rz",
+    "s",
+    "t",
+    "x",
+    "y",
+    "z"
+]
+
+
+class QiskitToQirBuilder(QirBuilder):
+    """Qiskit to QIR builder."""
+    def __init__(self, module: str):
```

Comment:
Should this allow for a default? I can imagine for a lot of usecases, there's only one module such that providing a reasonable default may help users out.

Reply:
This builder likely won't be used directly by users, so I am not sure we'd need to specify it. That being said, it is publically facing so might as well add a default.